cmd/agent/main.go

```diff
@@ -139,50 +107,69 @@ func main() {
 	}
 
 	requestString := MakeString(serverAddress)
-	var metricValueF64 float64
 	go GetMetrics(&mapMetrics, &PollCount, time.Duration(pollInt), &mutex)
 
 	for {
 		time.Sleep(time.Duration(reportInt) * time.Second)
 		mutex.RLock()
-
+		i := 0
+		metrics := make([]data.Metrics, 29)
 		for metricName, metricValue := range mapMetrics {
-			metricData := Metrics{}
+			metricData := data.Metrics{}
 			metricData.ID = metricName
 			metricData.MType = "gauge"
-			metricValueF64, err = strconv.ParseFloat(fmt.Sprint(metricValue), 64)
+			metricValueF64, err := strconv.ParseFloat(fmt.Sprint(metricValue), 64)
 			if err != nil {
 				fmt.Printf("Error while parsing metric %s: %s", metricName, err)
 			}
 			metricData.Value = &metricValueF64
-			compressedMetric, err := metricData.Compress()
-			if err != nil {
-				fmt.Println(err)
-			}
-			_, err = client.R().
-				SetHeader("Content-Type", "application/json").
-				SetHeader("Content-Encoding", "gzip").
-				SetHeader("Accept-Encoding", "gzip").
-				SetBody(compressedMetric).
-				Post(requestString)
-			if err != nil {
-				fmt.Printf("Error while sending metric %s: %s\n", metricName, err)
-			}
+			metrics[i] = metricData
+			i += 1
 		}
-		metricData := Metrics{}
+		metricData := data.Metrics{}
 		metricData.ID = "PollCount"
 		metricData.MType = "counter"
 		metricData.Delta = &PollCount
+		metrics[i] = metricData
+
+		compressedMetrics, err := data.Compress(&metrics)
+		if err != nil {
+			fmt.Printf("Error while compressing data: %s", err)
+		}
+		fmt.Printf("Send metrics to server\n")
 		_, err = client.R().
 			SetHeader("Content-Type", "application/json").
-			SetBody(metricData).
+			SetHeader("Accept-Encoding", "gzip").
+			SetHeader("Content-Encoding", "gzip").
+			SetBody(compressedMetrics).
 			Post(requestString)
-		if err != nil {
-			fmt.Printf("Error while sending PollCounter for metric PollCount: %s", err)
-		} else {
-			PollCount = 0
+
+		for i := 0; i < 3; i++ {
+			if err == nil {
+				PollCount = 0
+				break
+			} else if !(retryerr.CheckErrorType(err)) {
+				fmt.Printf("Error while sending metrics: %s", err)
+				break
+			} else if retryerr.CheckErrorType(err) {
+				if i == 0 {
+					time.Sleep(1 * time.Second)
+				} else {
+					time.Sleep(time.Duration(i+2) * time.Second)
+				}
+			}
+			fmt.Printf("Send metrics to server\n")
+			_, err = client.R().
+				SetHeader("Content-Type", "application/json").
+				SetHeader("Accept-Encoding", "gzip").
+				SetHeader("Content-Encoding", "gzip").
+				SetBody(compressedMetrics).
+				Post(requestString)
 		}
 
+		if (err != nil) && retryerr.CheckErrorType(err) {
+			fmt.Printf("Network error while sending metrics: %s", err)
```

Comment:
```suggestion
			fmt.Printf("Network error while sending metrics: %s\n", err)
```
Но вообще на этом этапе разработки в коде уже не должно быть `fmt`. Все диагностические сообщения следует делать через logger.